sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeServiceAsyncClient.java

```diff
@@ -284,12 +285,18 @@ public String getAccountUrl() {
             .listBlobContainers(Transforms.toListBlobContainersOptions(options));
         /* We need to create a new PagedFlux here because PagedFlux extends Flux, but not all operations were
             overriden to return PagedFlux - so we need to do the transformations and recreate a PagedFlux. */
-        /* Note: pageSize is not passed in as a parameter since the underlying implementation of listBlobContainers
-           does not use the pageSize parameter. Passing it in will have no effect. */
-        return PagedFlux.create(() -> (continuationToken, pageSize) -> {
-            Flux<PagedResponse<BlobContainerItem>> flux = (continuationToken == null)
-                ? inputPagedFlux.byPage()
-                : inputPagedFlux.byPage(continuationToken);
+        return StoragePagedFlux.create(() -> (continuationToken, pageSize) -> {
+            Flux<PagedResponse<BlobContainerItem>> flux;
+            if (continuationToken != null && pageSize != null) {
+                flux = inputPagedFlux.byPage(continuationToken, pageSize);
```

Comment:
isn't `byPage` smart enough to deal with optional parameters ?

Reply:
No. If you pass a null continuationToken, it short circuits and returns an empty flux. We need to specifically use the overload which matches the values we have